featurebyte/worker/task/base.py

```diff
@@ -70,14 +71,29 @@ async def update_progress(self, percent: int, message: str | None = None) -> Non
             # publish to redis
             self.progress.put(progress_dict)
 
+    def get_payload_obj(self, payload_data: dict[str, Any]) -> TaskT:
```

Comment:
add this helper here as it makes sense that the `Task` object itself can construct an appropriately typed task payload object.